cogs/welcome.py

```diff
@@ -197,78 +228,101 @@ def __init__(self, bot):
         """Enables, and sets current channel as log channel."""
         self.loadSettings()
         try:
-            self.settings[ctx.message.author.server.id][self.key_welcomeLogChannel] = ctx.message.channel.id

```

Comment:
serverId = ctx.message.author.server.id

Reply:
True, I made those changes, thanks for the feedback!